eng/Versions.props

```diff
@@ -207,9 +207,9 @@
     -->
     <Analyzer_MicrosoftCodeAnalysisCSharpVersion>3.3.1</Analyzer_MicrosoftCodeAnalysisCSharpVersion>
     <Analyzer_MicrosoftCodeAnalysisCSharpWorkspacesVersion>3.3.1</Analyzer_MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisCommonVersion>4.2.0-2.22128.1</MicrosoftCodeAnalysisCommonVersion>
-    <MicrosoftCodeAnalysisCSharpVersion>4.2.0-2.22128.1</MicrosoftCodeAnalysisCSharpVersion>
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.2.0-2.22128.1</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
+    <MicrosoftCodeAnalysisCommonVersion>4.4.0-2.22464.20</MicrosoftCodeAnalysisCommonVersion>
```

Comment:
@jaredpar just wanted to confirm versioning with you.

For `main` (this PR), is `4.4.0-2.22464.20` (from [sdk](https://github.com/dotnet/sdk/blob/release/7.0.1xx/eng/Versions.props) appropriate?

Likewise, for `release/7.0`, would `4.3.0-2.final` (from [runtime](https://github.com/dotnet/runtime/blob/release/7.0/eng/Versions.props#L47-L51)) be appropriate?

Reply:
> For main (this PR), is 4.4.0-2.22464.20 (from [sdk](https://github.com/dotnet/sdk/blob/release/7.0.1xx/eng/Versions.props) appropriate?

What relesae are you updating for?

Reply:
Main targets 8.0 now, so I believe .NET 8 Preview 1? Honestly not too concerned about `main` at the moment, the focus is on getting the appropriate versioning for `release/7.0`.

Reply:
For `release/7.0` use version 4.4.0-3.22472.13. Make sure you have the following feed in your restore sources 

https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json